docs/adr/0007-azure-pipeline-security-model.md

```diff
@@ -0,0 +1,70 @@
+# Azure pipeline security model
+
+## Status
+
+Proposed
+
+
+## Context
+
+This ADR defines an approach for a least-privilege Azure DevOps Pipeline that supports the automated deployment, teardown and re-deployment of a complete Marain instance.
+
+The challenge here is to balance the varied benefits of having a self-sufficient, repeatable pipeline with the need for adequate security controls to minimise the risk of the pipeline becoming the source of a security incident.
+
+Such a deployment pipeline needs to be able to perform the following actions:
+
+1. Create resources in the target subscription
+1. Query AAD for existing applications & service principals
+1. Perform ARM role assignments in the target subscription
+1. Create and manage new AAD applications & service principals
+
```

Comment:
Since this list contains entries for discovering AAD apps and principals, and also creating them, then it should also mention assigning service principals into App Roles. (That's similar in concept to but quite different in mechanism from ARM role assignments.)

(This would conflict with a "least privilege" approach, of course. But so do other entries in this list. I'd say this is really a list describing the things you need to do to stand up an environment, all of which _could_ be done in the pipeline, but some of which would not be done in the pipeline if you applied the principal of least privilege. And we definitely need a list of all the things that need to be done, and this App Role assignment step belongs in that list, specifically in the subsection of that list containing the things you wouldn't do in the deployment pipeline if applying the principal of least privilege, but which you might want to do in the deployment pipeline for convenience.)

Reply:
In my head that was covered by the 'manage' part of 'create and manage AAD application', but it's worth calling-out explicitly.